### PR TITLE
Prac 5 code review request

### PR DESCRIPTION
**Pull request by Jimmyrynn**
Hi @callumjriley3 could you check my variable naming and general structure.

Thanks 👍

### CODE REVIEW

File: prac_05/state_names.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"""
+CP1404/CP5632 Practical
+State names in a dictionary
+File needs reformatting
+"""
+
+CODE_TO_NAME = {"QLD": "Queensland", "NSW": "New South Wales", "NT": "Northern Territory", "WA": "Western Australia",
```
Comment (callumjriley3) [inline prac_05/state_names.py:7]: Great dictionary name! Should the dictionary also include South Australia?

File: prac_05/state_names.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+while state_code != "":
+    if state_code in CODE_TO_NAME:
+        print(state_code, "is", CODE_TO_NAME[state_code])
+    else:
+        print("Invalid short state")
+    state_code = input("Enter short state: ").upper()
+
+# b.
+for state_code in CODE_TO_NAME:
+    print(f"{state_code:3} is {CODE_TO_NAME[state_code]}")
```
Comment (callumjriley3) [inline prac_05/state_names.py]: Would it be better if the 3 was a constant?

File: prac_05/state_names.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+# c.
+state_code = input("Enter short state: ").upper()
+while state_code != "":
+    try:
+        if state_code in CODE_TO_NAME:
```
Comment (callumjriley3) [inline prac_05/state_names.py]: Since you're using the try-exception structure, is the if statement still needed?

File: prac_05/hex_colours.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+COLOUR_NAME_TO_CODE = {"aqua": "#00ffff", "black": "#000000", "blue": "#0018a8", "brown": "#a52a2a",
+                       "burgundy": "#800020", "celadon": "#ace1af", "chocolate": "#d2691e",
+                       "citron": "#9fa91f", "copper": "#b87333", "cream": "#fffdd0"}
+
+print(COLOUR_NAME_TO_CODE)
+colour_name = input("Enter colour: ").lower()
+while colour_name != "":
+    if colour_name in COLOUR_NAME_TO_CODE:
```
Comment (callumjriley3) [inline prac_05/hex_colours.py]: If-else seems to work fine, could this be enhanced with exception handling?

File: prac_05/hex_colours.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+COLOUR_NAME_TO_CODE = {"aqua": "#00ffff", "black": "#000000", "blue": "#0018a8", "brown": "#a52a2a",
+                       "burgundy": "#800020", "celadon": "#ace1af", "chocolate": "#d2691e",
+                       "citron": "#9fa91f", "copper": "#b87333", "cream": "#fffdd0"}
+
+print(COLOUR_NAME_TO_CODE)
+colour_name = input("Enter colour: ").lower()
```
Comment (callumjriley3) [inline prac_05/hex_colours.py:6]: Nice consistent naming between colour_name and COLOUR_NAME_TO_CODE :)

File: prac_05/word_occurrences.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+"""
+Word Occurrences
+Estimate: 10 minutes
+Actual: 22 minutes
+"""
+word_to_frequency = {}
+# text = input("Enter Text: ")
+text = "this is a collection of words of nice words this is a fun thing it is"
+words = text.split()
```
Comment (callumjriley3) [inline prac_05/word_occurrences.py]: Could this be combined with the for statement to create something like "for word in text.split()"?

File: prac_05/word_occurrences.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+for word in words:
+    frequency = word_to_frequency.get(word, 0)
+    word_to_frequency[word] = frequency + 1
```
Comment (callumjriley3) [inline prac_05/word_occurrences.py]: I could be wrong, but would it work if these were combined as one line, something like:
word_to_frequency[word] = word_to_frequency.get(word, 0) + 1

File: prac_05/word_occurrences.py [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+words = list(word_to_frequency.keys())
+words.sort()
+longest_word = max((len(word) for word in words))
```
Comment (callumjriley3) [inline prac_05/word_occurrences.py]: As you're finding the maximum length, is the name longest_word misleading?

File: prac_05/emails.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        valid_name = input(f"Is your name {name.title()} (Y/n) ").lower()
+        if valid_name != 'y' and valid_name != "":
+            name = input("Name: ")
+        email_to_name[user_email] = name
+        user_email = input("Email: ")
+    for user_email, name in email_to_name.items():
+        print(f"{name} ({user_email})")
+
+
+def get_name_from_email(user_email):
```
Comment (callumjriley3) [inline prac_05/emails.py:22]: Good function name :)

File: prac_05/emails.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+"""
+
+
+def main():
+    """Print names from user emails."""
+    email_to_name = {}
+    user_email = input("Email: ")
+    while user_email != '':
+        name = get_name_from_email(user_email)
+        valid_name = input(f"Is your name {name.title()} (Y/n) ").lower()
```
Comment (callumjriley3) [inline prac_05/emails.py]: Could the variable called valid_name be misleading as the user isn't entering a name but a **selection**?

File: prac_05/wimbledon.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def process_records(records):
+    """Create a count of players and championships won."""
+    champion_to_count = {}
+    countries = set()
+    for record in records:
+        countries.add(record[INDEX_COUNTRY])
+        try:
+            champion_to_count[record[INDEX_CHAMPION]] += 1
```
Comment (callumjriley3) [inline prac_05/wimbledon.py]: Would lines 25 & 27 work better using champion_to_count.get (similar to how you used it in word_occurrences.py)?

File: prac_05/wimbledon.py [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+"""
+FILENAME = "wimbledon.csv"
+INDEX_COUNTRY = 1
+INDEX_CHAMPION = 2
+
+
+def main():
+    """Process data from a file and format accordingly."""
+    records = get_records(FILENAME)
+    champion_to_count, countries = process_records(records)
```
Comment (callumjriley3) [inline prac_05/wimbledon.py:13]: Would the dictionary name champion_to_count be easier to understand if it was something like "champion_to_number_of_wins"? (I understand the solutions for wimbledon.py also used champion_to_count)

File: prac_05/wimbledon.py [hunk 13]
```diff
@@ -0,0 +1,51 @@
+"""
+Word Occurrences
```
Comment (callumjriley3) [inline prac_05/wimbledon.py]: Was this unintentional? (Don't worry, I've left text in docstrings that was supposed to be removed before!)

File: prac_05/state_names.py (hunk 1 above)
Comment (Jimmyrynn) [reply prac_05/state_names.py:7]: Good catch 👍

File: prac_05/state_names.py (hunk 2 above)
Comment (Jimmyrynn) [reply prac_05/state_names.py]: 👍

File: prac_05/state_names.py (hunk 3 above)
Comment (Jimmyrynn) [reply prac_05/state_names.py]: That's true. Fixed

File: prac_05/hex_colours.py (hunk 4 above)
Comment (Jimmyrynn) [reply prac_05/hex_colours.py]: Good Catch 👍

File: prac_05/word_occurrences.py (hunk 6 above)
Comment (Jimmyrynn) [reply prac_05/word_occurrences.py]: That's a good point.

File: prac_05/word_occurrences.py (hunk 7 above)
Comment (Jimmyrynn) [reply prac_05/word_occurrences.py]: Yep that works 😄

File: prac_05/word_occurrences.py (hunk 8 above)
Comment (Jimmyrynn) [reply prac_05/word_occurrences.py]: Good observation

File: prac_05/wimbledon.py (hunk 12 above)
Comment (Jimmyrynn) [reply prac_05/wimbledon.py:13]: Yep 😄

**Comment by Jimmyrynn** [comment]
Thanks for the feedback, very helpful 👍

**Pull request merged**